### PR TITLE
Run Rust tests in CI

### LINKED ISSUE #80: Run Rust tests in CI

### PR DESCRIPTION
**Pull request by tiziano88**
- Add run_tests script to run tests on Rust lib and examples.
- Make run_tests run on Travis.
- Add Cargo.toml file for lib workspace, so that dependencies between
the oak and oak_derive crates are shared.
- Fix tests; note that in Rust, doc comments with examples that do not
compile count as failed tests too.

Fixes #80

---
This change is [Reviewable](https://reviewable.io/reviews/project-oak/oak/81)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,5 +5,6 @@ services:
 
 script:
   - ./scripts/docker_run ./scripts/check_formatting
+  - ./scripts/docker_run ./scripts/run_tests
```
Comment (tiziano88) [inline .travis.yml]: test comment -- please ignore

**Comment by anghelcovici** [comment]
This seems to break the CI due to some permission issues. Maybe worth trying to rebase on my changes to see if it fixes it.

**Comment by tiziano88** [comment]
Yes, this will fail until #86 is merged, so please review under the assumption that it actually works :) ofc this will not be merged until it actually passes.

**Comment by tiziano88** [comment]
FYI this is now passing, PTAL @anghelcovici .

**Pull request merged**